server/src/main/java/com/objectcomputing/checkins/services/checkin_notes/CheckinNoteServicesImpl.java

```diff
@@ -43,25 +66,42 @@ public CheckinNote save(CheckinNote checkinNote) {
 
     @Override
     public CheckinNote read(@NotNull UUID id) {
-        return checkinNoteRepository.findById(id).orElse(null);
+        String workEmail = securityService!=null ? securityService.getAuthentication().get().getAttributes().get("email").toString() : null;
+        MemberProfile currentUser = workEmail!=null? currentUserServices.findOrSaveUser(null, workEmail) : null;
+        Boolean isAdmin = securityService!=null ? securityService.hasRole(RoleType.Constants.ADMIN_ROLE) : false;
+        CheckinNote checkInNoteResult =  checkinNoteRepository.findById(id).orElse(null);
+        validate(checkInNoteResult == null, "Invalid checkin note id %s", id);
+        if(!isAdmin) {
```

Comment:
This can be accomplished with the @Secured annotation as is done on the create and update operations

Reply:
I didn't do that because on the controller I used IS_AUTHENTICATED annotation and then on create/update I did use admin and pdl because only they can create a note. But on /read end point even a member can read a checkin note and that's why this validation is there to check if its an admin then don't verify anything else and let them read it or else do other validations to make sure only pdl of the subject or the subject can read the notes.

Reply:
Oh, I see. I read the code too quickly and missed what was going on.

I'm going to say in that case that this should throw a different exception that will be caught by the controller and return a 401, rather than 400.

Reply:
I was reading about this this morning and it turns out I was wrong. This should return a 403, not a 401.
https://tools.ietf.org/html/rfc7231#section-6.5.3